modules/Makefile.kubernetes

```diff
@@ -85,13 +85,34 @@ kubernetes\:replace-controller:
 kubernetes\:replace:
 	@$(SELF) kubernetes:replace-service kubernetes:replace-controller
 
+# (private) Create a new deployment
+kubernetes\:create-deployment:
+	$(call kubectl_create,deployment)
+
+# (private) Delete a deploymentt
+kubernetes\:delete-deployment:
+	$(call kubectl_delete,deployment)
+
+# (private) Update existing controller using rolling-update or create a new controller; do not notify datadog
+kubernetes\:deploy-deployment:
```

Comment:
Okay, can see your point. I think it depends on the angle of approach.

We have a select few verbs that we use like `create`, `delete, `replace`, and `deploy`. 

The `deploy` verb is one that does not conflict with `kubectl` actions and was chosen to mean "take the best course of action in order to accomplish some task using `kubectl` operations". 
e.g.

`kubernetes:deploy-controller`: figures out if the `controller` resource exists. if it exists, then it does a `rolling-update`, otherwise it does a `create`.

`kubernetes:deploy-service`: figures out if the `service` resource has changed or does not exist. if 
it has changed or does not exist, then it does a `kubernetes:replace` (which calls `delete` then `create`), otherwise it does nothing.

`kubernetes:deploy-deployment`: figures out if the `deployment` resource exists. if it exists, then it updates it by calling `apply` on the resource, otherwise it calls `create` instantiate a new one.

Lastly, there's the `kubernetes:deploy` target that figures out which of the `deploy-<something>` targets are applicable and calls them in the appropriate order.

I think `kubernetes:deploy` is all that a a user should care about which is why the other targets are "private" (as in not available in `make help`), but named consistently for maintainability of the code. 

It's a lose/lose. Feelings won't be hurt either way. =)